build/common.sh

```diff
@@ -569,6 +578,30 @@ function kube::release::package_server_tarballs() {
   done
 }
 
+# This will take binaries that run on master and creates Docker images
+# that wrap the binary in them. (One docker image per binary)
+function kube::release::create_docker_images_for_server() {
+  # Create a sub-shell so that we don't pollute the outer environment
+  (
+    local binary_name;
+    for binary_name in "${KUBE_DOCKER_WRAPPED_BINARIES[@]}"; do
+      echo "+++ Building docker image: ${binary_name}";
+      local docker_file_path="$1/Dockerfile";
+      local binary_file_path="$1/${binary_name}";
+      if [ -f ${docker_file_path} ]; then
+        rm ${docker_file_path};
+      fi;
+      printf " FROM scratch \n ADD ${binary_name} /${binary_name} \n ENTRYPOINT [ \"/${binary_name}\" ]\n" >> ${docker_file_path};
+      local md5_sum=$(md5sum ${binary_file_path} | awk '{print $1}')
+      local docker_image_tag=gcr.io/google_containers/$binary_name:$md5_sum
+      docker build -t "${docker_image_tag}" ${1};
```

Comment:
this doesn't work unless the machine you are provisioning on has docker installed by default - i don't think this is the case for all salt-based providers.  this also doesn't make sense if we have salt scripts that are later responsible for installing docker as part of salt provisioning.